tests/tests/test_vsock_vm.py

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env python3
+#
+# Copyright 2024 Canonical Ltd.
+#
+# This program is free software: you can redistribute it and/or modify it
+# under the terms of the GNU General Public License version 3, as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranties of MERCHANTABILITY,
+# SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+import os
+import subprocess
+import threading
+import time
+
+import Qemu
+
+# Global Variables
+script_path=os.path.dirname(os.path.realpath(__file__))
+guest_cid=25
```

Comment:
Is there a way to avoid hard-coding this? Libvirt vsock has this `cid auto='yes'` feature that we use [here](https://github.com/canonical/tdx/blob/noble-24.04/guest-tools/trust_domain.xml.template#L50-L53) to ensure we don't accidentally ask for a cid that is already allocated.

Reply:
I had searched for a way to do that automagically for qemu, but didn't find anything.  The only way to do it is to "try" and repeat.  I think given the way this will be often tested, its fine.   Most of the time these tests will run in a clean environment I would expect.